text/0087-klt-traceid-propagation.md

```diff
@@ -0,0 +1,21 @@
+# KEP0087 - Lifecycle Toolkit Parent Trace ID in Keptn Apps 
+
+**State: DRAFTING**
+
+## Motivation
+When promoting apps to other instances, an end user would like to see the whole flow across the stages in a single trace. This KEP proposes the usage of a parent trace ID, to create links between traces across stages.
+
+## Current Design
+Currently, the Keptn Lifecycle Toolkit (Controller) starts a new trace whenever a new version/revision of a KeptnAppVersion is created and there is no way to pass over a TraceId to an Application. As a result, the Trace ends when an application is deployed and there is no Link to the next stage. 
+
+## Proposed Design
+This KEP proposes to add a new field to the KeptnApp CRD called `parentTraceId`. This field is optional and can be used to pass over a TraceId to the KeptnApp. The main use case for this field is to pass over the TraceId of the previous stage to the next stage, when promoting an application and therefore keep an end to end view of the whole flow.
```

Comment:
I think `parent` is a bad name since it is not really a parent-child relationship. I would suggest using `previous`.

```suggestion
This KEP proposes to add a new field to the KeptnApp CRD called `previousTraceId`. This field is optional and can be used to pass over a TraceId to the KeptnApp. The main use case for this field is to pass over the TraceId of the previous stage to the next stage when promoting an application and therefore keep an end to end view of the whole flow.
```